x-pack/filebeat/input/s3/input.go

```diff
@@ -0,0 +1,449 @@
+	if err != nil {
+		panic(err)
+	}
+}
+
+// Input is a input for s3
+type Input struct {
+	outlet    channel.Outleter
+	config    config
+	awsConfig awssdk.Config
+	logger    *logp.Logger
+	close     chan struct{}
+}
+
+type s3Info struct {
+	name   string
+	key    string
+	region string
+	arn    string
+}
+
+type sqsMessage struct {
+	Records []struct {
+		EventSource string `json:"eventSource"`
+		AwsRegion   string `json:"awsRegion"`
+		EventName   string `json:"eventName"`
+		S3          struct {
+			Bucket struct {
+				Name string `json:"name"`
+				Arn  string `json:"arn"`
+			} `json:"bucket"`
+			Object struct {
+				Key string `json:"key"`
+			} `json:"object"`
+		} `json:"s3"`
+	} `json:"Records"`
+}
+
+// NewInput creates a new s3 input
+func NewInput(cfg *common.Config, outletFactory channel.Connector, context input.Context) (input.Input, error) {
+	cfgwarn.Beta("s3 input type is used")
+
+	logger := logp.NewLogger(inputName)
+
+	config := defaultConfig
+	if err := cfg.Unpack(&config); err != nil {
+		return nil, errors.Wrap(err, "failed unpacking config")
+	}
+
+	outlet, err := outletFactory(cfg, context.DynamicFields)
+	if err != nil {
+		return nil, err
+	}
+
+	err = config.validate()
+	if err != nil {
+		return nil, errors.Wrapf(err, "validation for s3 input config failed: config = %v", config)
+	}
+
+	awsConfig, err := getAWSCredentials(config)
+	if err != nil {
+		return nil, errors.Wrap(err, "getAWSCredentials failed")
+	}
+
+	p := &Input{
+		outlet:    outlet,
+		config:    config,
+		awsConfig: awsConfig,
+		logger:    logger,
+		close:     make(chan struct{}),
+	}
+
+	return p, nil
+}
+
+func (c *config) validate() error {
+	if c.VisibilityTimeout < 0 || c.VisibilityTimeout.Hours() > 12 {
+		return errors.New("visibilityTimeout is not defined within the " +
+			"expected bounds")
+	}
+
+	if len(c.QueueURLs) == 0 {
+		return errors.New("no SQS queueURLs are configured")
+	}
+	return nil
+}
+
+func getAWSCredentials(config config) (awssdk.Config, error) {
+	// Check if accessKeyID and secretAccessKey is given from configuration
+	if config.AccessKeyID != "" && config.SecretAccessKey != "" {
+		awsConfig := defaults.Config()
+		awsCredentials := awssdk.Credentials{
+			AccessKeyID:     config.AccessKeyID,
+			SecretAccessKey: config.SecretAccessKey,
+		}
+		if config.SessionToken != "" {
+			awsCredentials.SessionToken = config.SessionToken
+		}
+
+		awsConfig.Credentials = awssdk.StaticCredentialsProvider{
+			Value: awsCredentials,
+		}
+		return awsConfig, nil
+	}
+
+	// If accessKeyID and secretAccessKey is not given, then load from default config
+	// Please see https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-profiles.html
+	// with more details.
+	return external.LoadDefaultAWSConfig(
+		external.WithSharedConfigProfile(config.ProfileName),
+	)
+}
+
+// Run runs the input
+func (p *Input) Run() {
+	p.logger.Debugf("s3", "Run s3 input with queueURLs: %+v", p.config.QueueURLs)
```

Comment:
This not a valid printf style message. The leading `s3` argument looks to be the problem because the logger is already named `s3`.